pkg/functiontest/synthlint_test.go

```diff
@@ -0,0 +1,247 @@
+	Cache    string `eno_key:"cache"`
+	Storage  string `eno_key:"storage"`
+	Network  string `eno_key:"network"`
+}
+
+type TestInputsEmpty struct {
+	Field1 string `json:"field1"`
+	Field2 string `json:"field2"`
+}
+
+func TestInputsMatchSynthesizerRefs_StrictMode_Success(t *testing.T) {
+	// Create temporary synthesizer.yaml with exact matching refs
+	synthesizerPath := createTempSynthesizer(t, []string{"database", "cache", "storage"})
+	defer os.Remove(synthesizerPath)
+
+	// Test with exact match - should pass
+	inputs := TestInputsComplete{}
+
+	// This should not fail in strict mode since all keys match exactly
+	InputsMatchSynthesizerRefs(t, inputs, synthesizerPath, KeyMatchStrict)
+}
+
+func TestInputsMatchSynthesizerRefs_StrictMode_FailureMissingEnoKey(t *testing.T) {
+	// Create synthesizer.yaml with more refs than eno_keys
+	synthesizerPath := createTempSynthesizer(t, []string{"database", "cache", "network"})
+	defer os.Remove(synthesizerPath)
+
+	// Test with missing eno_key - should fail in strict mode
+	inputs := TestInputsPartial{} // only has database and cache
+
+	// Create a mock testing.T to capture the failure
+	mockT := &mockTestingT{}
+	InputsMatchSynthesizerRefs(mockT, inputs, synthesizerPath, KeyMatchStrict)
+
+	// Verify that the test failed
+	assert.True(t, mockT.failed, "Expected test to fail in strict mode when eno_keys are missing")
+	assert.Contains(t, mockT.errorMsg, "network", "Expected error message to mention missing 'network' key")
+}
+
+func TestInputsMatchSynthesizerRefs_StrictMode_FailureExtraEnoKey(t *testing.T) {
+	// Create synthesizer.yaml with fewer refs than eno_keys
+	synthesizerPath := createTempSynthesizer(t, []string{"database", "cache"})
+	defer os.Remove(synthesizerPath)
+
+	// Test with extra eno_key - should fail in strict mode
+	inputs := TestInputsComplete{} // has database, cache, and storage
+
+	mockT := &mockTestingT{}
+	InputsMatchSynthesizerRefs(mockT, inputs, synthesizerPath, KeyMatchStrict)
+
+	// Verify that the test failed
+	assert.True(t, mockT.failed, "Expected test to fail in strict mode when extra eno_keys exist")
+	assert.Contains(t, mockT.errorMsg, "storage", "Expected error message to mention extra 'storage' key")
+}
+
+func TestInputsMatchSynthesizerRefs_RelaxedMode_Success(t *testing.T) {
+	// Create synthesizer.yaml with more refs than eno_keys
+	synthesizerPath := createTempSynthesizer(t, []string{"database", "cache", "storage", "network"})
+	defer os.Remove(synthesizerPath)
+
+	// Test with subset of refs - should pass in relaxed mode
+	inputs := TestInputsPartial{} // only has database and cache
+
+	// This should not fail in relaxed mode since eno_keys are a subset of refs
+	InputsMatchSynthesizerRefs(t, inputs, synthesizerPath, KeyMatchRelaxed)
+}
+
+func TestInputsMatchSynthesizerRefs_RelaxedMode_FailureMissingRef(t *testing.T) {
+	// Create synthesizer.yaml with fewer refs than eno_keys
+	synthesizerPath := createTempSynthesizer(t, []string{"database", "cache"})
```

Comment:
personally I'd prefer a couple of test fixtures instead of this helper. Nice to read the fixtures directly when debugging failing tests

Reply:
By fixtures you mean checked in files right? Thats interestin y instict was to go the opposite way make InputsMatchSynthesizerRefs take a reader and never goto disk at all.

Reply:
Maybe I am just burned from all the file regeneration in ccp/control-plane-core. 

AI seems happy to generate thos yamls and it does make the code shorter. Pushing that soon